tools/clang/test/HLSLFileCheck/hlsl/intrinsics/atomic/atomic_overload.hlsl

```diff
@@ -14,6 +14,7 @@
 // RUN: %dxc -no-warnings -T cs_6_6 -DTYPE=uint64_t  %s | %FileCheck %s -check-prefixes=INTCHK
 // RUN: %dxc -no-warnings -T cs_6_0 -DTYPE=int  %s | %FileCheck %s -check-prefixes=INTCHK
 // RUN: %dxc -no-warnings -T cs_6_0 -DTYPE=uint  %s | %FileCheck %s -check-prefixes=INTCHK
+// REQUIRES: dxilver-1.6
```

Comment:
Shoot, this is a problem.  We have some multi-run-line tests that vary the version and require different versions here for certain run lines.  Additionally, we actually do the requirement check based on the shader model in the dxc command line as well.  If the compiler or (more likely) validator do not support the target shader model, the run line would be skipped.

How does that fit into this model?

Reply:
In LIT, each test file is treated as an individual test case, and requirements are applied to the test case as a whole. If the test case has multiple `RUN` lines, it is still one test case. For cases where the same input is processed differently or has different requirements based on options, the general approach is to break the test into multiple test cases.

The different test cases can either share the same input file (by replacing `%s` in the `RUN` line with the path to the input file, or they can use an include directive to include the underlying source.

I can rework this patch to adjust.

Reply:
I guess the crux of the issue is whether this work is being done to support running in a different way that is not capable of deciding to skip (without failing) the remainder of a run line on the basis of something that needs to be run in that line (not a simple requirement check for the file).

I bring this up because there are implicit `CheckDxilVer` calls in `%dxc` and `%link` commands, so these commands may also want to cause a skip of the remainder of the run commands without a failure.

Reply:
The way LIT works, is those lines are literally shell commands with substitutions and pipes. If a command in the chain fails, the test fails, not skips to the next test. Migrating to a "REQUIRES" model, means making those requirements explicit.

Reply:
> Migrating to a "REQUIRES" model, means making those requirements explicit.

In that case, I think there's more work here than you think.  Due to the implicit check, you won't have picked up all the extra cases that would need `REQUIRES:` but didn't use `%dxilver`.  This may reveal more files with multiple RUN lines that don't have the same requirements for all the lines (the other issue being discussed here).